homeassistant/components/shelly/utils.py

```diff
@@ -237,3 +238,23 @@ def get_model_name(info: dict[str, Any]) -> str:
         return cast(str, MODEL_NAMES.get(info["model"], info["model"]))
 
     return cast(str, MODEL_NAMES.get(info["type"], info["type"]))
+
+
+def get_rpc_entity_name(
+    device: RpcDevice, key: str, description: str | None = None
+) -> str:
+    """Naming for RPC based switch and sensors."""
+    entity_name: str | None = device.config[key].get("name")
+
+    if entity_name is None:
+        entity_name = f"{get_rpc_device_name(device)} {key.replace(':', '_')}"
+
+    if description:
+        return f"{entity_name} {description}"
+
+    return entity_name
+
+
+def get_device_entry_gen(entry: ConfigEntry) -> int:
+    """Return the device generation from config entry."""
+    return int(entry.data.get("gen", 1))
```

Comment:
How come that we need to cast this to an `int` ? When do we store it as a string?